### PR TITLE
[#4] update benchmarks

### PR DESCRIPTION
**Pull request by elfenpiff**
## Notes for Reviewer

With v0.4 we refactored the code, benchmarks and improved the overall performance of iceoryx2. This PR updates the benchmark results and diagrams.

## Pre-Review Checklist for the PR Author

1. [x] Add sensible notes for the reviewer
1. [x] PR title is short, expressive and meaningful
1. [x] Relevant issues are linked in the [References](#references) section
1. [x] Every source code file has a copyright header with `SPDX-License-Identifier: Apache-2.0 OR MIT`
1. [x] Branch follows the naming format (`iox2-123-introduce-posix-ipc-example`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`[#123] Add posix ipc example`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Assign PR to reviewer
1. [x] All checks have passed (except `task-list-completed`)

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx2/blob/main/doc/release-notes/iceoryx2-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Unit tests have been written for new behavior
- [x] Public API is documented
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [ ] All open points are addressed and tracked via issues

## References

Relates to #4

### CODE REVIEW

File: internal/plots/benchmark_architecture.svg
Comment (orecham) [inline internal/plots/benchmark_architecture.svg:1]: Would it be useful to keep older versions too e.g. in an `archive` directory ? There were some nice gains this release.

File: internal/plots/benchmark_architecture_os_comparision.dat
Comment (orecham) [inline internal/plots/benchmark_architecture_os_comparision.dat:1]: Or maybe even add a plot showing the trend of this table across releases?

**Review by orecham — COMMENTED** [review]
Some questions.

File: internal/plots/benchmark_architecture_os_comparision.dat [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
-  "AMD Ryzen 9 5950X"           "Windows"   205
-  "AMD Ryzen 9 5950X"           "Linux"     168                     173
+# System                        OS          Latency Same Core (HyperThreading)      Latency Different Cores
+  "Raspberry Pi 4B"             "Linux"     774                                     0
+  "Intel i7-13700H"             "Linux"     92                                      326
+  "Intel i7-8565U"              "Linux"     270                                     263
+  "Intel i7-7500U"              "Linux"     276                                     0
+  "AMD Ryzen 7 7840S"           "Linux"     109                                     222
+  "AMD Ryzen 9 5950X"           "Windows"   205                                     0
+  "AMD Ryzen 9 5950X"           "Linux"     168                                     173
```
Comment (elBoberido) [inline internal/plots/benchmark_architecture_os_comparision.dat:8]: Have you tried something like this?
```suggestion
  "AMD Ryzen 9 5950X"           "Linux [Intra-Core]"     168
  "AMD Ryzen 9 5950X"           "Linux [Inter-Core]"     173
```

File: internal/plots/benchmark_architecture.svg
Comment (elBoberido) [reply internal/plots/benchmark_architecture.svg:1]: Ideally, these numbers would be obtained on the CI with each PR

see also https://github.com/eclipse-iceoryx/iceoryx/issues/137

File: internal/plots/benchmark_mechanism_comparision_rp4.dat [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
-    16      0.875
-    32      0.875
-    64      0.875
-    128     0.875
-    256     0.875
-    512     0.875
-    1024    0.875
-    2048    0.875
-    4096    0.875
+    0.064   0.922
```
[the remark this answers is not in the document]
Comment (elBoberido) [reply internal/plots/benchmark_mechanism_comparision_rp4.dat:4]: I would guess we need some warmup phase where we make the OS spin up the CPUs. iceoryx2 is so fast that the OS keeps the CPU in a powersafe mode for some milliseconds.

[the remark this answers is not in the document]
Comment (elfenpiff) [reply internal/plots/benchmark_mechanism_comparision_rp4.dat:4]: Yes this and also with the insane performance of iceoryx2 background tasks like a ssh connection, fish shell and whatever seem to start to affect the benchmark ... especially for weaker systems like the raspberry pi.

File: internal/plots/benchmark_architecture_os_comparision.dat (hunk 1 above)
Comment (elfenpiff) [reply internal/plots/benchmark_architecture_os_comparision.dat:8]: Yes, but it is for me not directly obvious what inter and intra core means. So we need an explanation inside the diagram with intra-core means hyperthreading and one core and inter core means different physical cpu cores but on the same CPU.

The intention of this PR for the moment was to just update the numbers. We can add this before the v0.4 release but I would do this in a separate PR if you do not mind.

File: internal/plots/benchmark_architecture.svg
Comment (elfenpiff) [reply internal/plots/benchmark_architecture.svg:1]: I added an archive folder and stored the overview file of v0.3 there. Additionally, I created a gnuplot script that presents the gains we made.

File: internal/plots/benchmark_architecture_os_comparision.dat
Comment (elfenpiff) [reply internal/plots/benchmark_architecture_os_comparision.dat:1]: The graph is done and I will add it in the release notes.

**Review by orecham — APPROVED** [review]

File: internal/plots/benchmark_architecture_os_comparision.dat (hunk 1 above)
Comment (elBoberido) [reply internal/plots/benchmark_architecture_os_comparision.dat:2]: No problem, we can also just keep it as is.

**Review by orecham — APPROVED** [review]

**Pull request merged**